Build error: 'ApiResult<TResult>' does not contain a constructor that takes 3 arguments

Error: D:\a\Library\Library\WebLib\Helpers\WebApiHelper.cs(14,12): error CS1729: 'ApiResult<TResult>' does not contain a constructor that takes 3 arguments [D:\a\Library\Library\WebLib\WebLib.csproj]

--- a/WebLib/Results/ApiResult.Generic.cs
+++ b/WebLib/Results/ApiResult.Generic.cs
@@ -6,7 +6,10 @@ namespace Library.Web.Results;
 
 public class ApiResult<T> : ApiResult, IApiResult<T>
 {
-    protected ApiResult(int? httpStatusCode = null, string? message = null, T? value = default)
+    public ApiResult(int? httpStatusCode = null, string? message = null, T? value = default)
+        : base(httpStatusCode, message) => this.Value = value;
+    
+    public ApiResult(HttpStatusCode? httpStatusCode = null, string? message = null, T? value = default)
         : base(httpStatusCode, message) => this.Value = value;
 
     public void Deconstruct(out int? statusCode, out string? message, out T? value)

